src/clr-angular/data/datagrid/providers/selection.spec.ts

```diff
@@ -311,6 +313,18 @@ export default function(): void {
                     expect(selectionInstance.isSelected(items[5])).toBe(true);
                 });
 
+                it("does not search for the selection when the items array is undefined", () => {
+                    const emptyItems = new Items(filtersInstance, sortInstance, pageInstance);
+                    const selection: Selection = new Selection(emptyItems, filtersInstance);
+
+                    spyOn(emptyItems, "trackBy");
+
+                    expect(selection.currentSingle).toBeUndefined();
+                    selection.currentSingle = items[2];
+
+                    expect(emptyItems.trackBy).not.toHaveBeenCalled();
+                });
```

Comment:
This test works for me, though the name is still strange. "does not apply trackBy to single selection with no items" or something is more clear to me.